From 9111bce4b7eb7fc56e6ffe7b0006fe809de176f1 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Fri, 16 Oct 2020 18:30:38 +0530
Subject: [PATCH] BN_bn2binpad api is not found even if <openssl/bn.h> is
 included

This needs to be revisited, just declaring it again as a workaround.

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 lib/tpm2_openssl.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/tpm2_openssl.c b/lib/tpm2_openssl.c
index 6d29c0d4..6797dd01 100644
--- a/lib/tpm2_openssl.c
+++ b/lib/tpm2_openssl.c
@@ -13,6 +13,9 @@
 #include "tpm2_alg_util.h"
 #include "tpm2_openssl.h"
 
+/* This is a weird warning, include <openssl/bn.h> didn't work */
+extern int BN_bn2binpad(const BIGNUM *a, unsigned char *to, int tolen);
+
 /* compatibility function for OpenSSL versions < 1.1.0 */
 #if OPENSSL_VERSION_NUMBER < 0x10100000L
 static int BN_bn2binpad(const BIGNUM *a, unsigned char *to, int tolen) {
-- 
2.17.1

